src/plugins/workspace/server/saved_objects/saved_objects_wrapper_for_check_workspace_conflict.ts

```diff
@@ -49,6 +68,16 @@ export class WorkspaceConflictSavedObjectsClientWrapper {
       options: SavedObjectsCreateOptions = {}
     ) => {
       const { workspaces, id, overwrite } = options;
+
+      if (workspaces?.length && (this.isDataSourceType(type) || this.isConfigType(type))) {
+        // For 2.14, data source can only be created without workspace info
+        // config can not be created inside a workspace
+        throw SavedObjectsErrorHelpers.decorateBadRequestError(
+          new Error(`'${type}' is not allowed to create in workspace.`),
```

Comment:
should it be the following?
```suggestion
          new Error(`'${type}' is not allowed to be created in workspace.`),
```